rust/hyperlane-base/src/contract_sync/cursor.rs

```diff
@@ -98,7 +356,15 @@ impl<I: Indexer> SyncBlockRangeCursor for RateLimitedSyncBlockRangeCursor<I> {
         Ok((from, to, eta))
     }
 
-    fn backtrack(&mut self, start_from: u32) {
-        self.from = u32::min(start_from, self.from);
+    async fn update(&mut self, _: Vec<(T, LogMeta)>) -> eyre::Result<()> {
+        // Store a relatively conservative view of the high watermark, which should allow a single watermark to be
+        // safely shared across multiple cursors, so long as they are running sufficiently in sync
```

Comment:
What happens if they don't run in sync?

Reply:
If two tasks share a high watermark, then task A may write a watermark that is too high for task B.

The scraper shares a watermark between the delivery and gas payment sync tasks. I didn't want to change the DB schema.